tests/qunit/adapter.js

```diff
@@ -0,0 +1,153 @@
+// Adapted from https://github.com/karma-runner/karma-qunit/blob/master/src/adapter.js, in
+// order to allow us to use our legacy version of QUnit (we're on v1.11.0pre, which isn't
+// available on NPM).
+//
+// TODO: We should update to a more recent version of QUnit, or we migrate to Mocha or
+//       something similar.
+
+// MIT licensed: https://github.com/karma-runner/karma-qunit/blob/master/LICENSE
```

Comment:
Normally you'd use the [`karma-qunit`](https://github.com/karma-runner/karma-qunit) package to manage the running of QUnit tests with Karma, but it assumes that QUnit has been installed as an NPM module. The version of QUnit we're using isn't available on NPM, and it has some specific APIs that aren't in the `1.0.0` version that is on NPM. So in order to avoid changing lots of the test files, I've slightly customised the adapter.

Ideally we should update to the most recent version of QUnit, or switch to Mocha/Jasmin/etc. I figured we could put that through as a separate PR though.